test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Options/PackageSourceMappingPageTests.cs

```diff
@@ -0,0 +1,145 @@
+
+            _packageSources =
+            [
+                new PackageSource(sourceUrl1, sourceName1, isEnabled: true),
+                // nonExistantSourceName is not added here intentionally.
+                new PackageSource(sourceUrl3, sourceName3, isEnabled: true)
+            ];
+
+            // Configure Unified Settings input to modify an existing package source mapping.
+
+            // Unchanged package source mapping.
+            Dictionary<string, object> packageSourceDictionary1 = new Dictionary<string, object>();
+            packageSourceDictionary1[PackageSourceMappingPage.MonikerPackageId] = unitTestingSourceMapping1.Patterns.Single().Pattern;
+            packageSourceDictionary1[PackageSourceMappingPage.MonikerSourceNames] = new List<string>() { unitTestingSourceMapping1.Key };
+
+            // nonExistantSourceName is not known to Unified Settings, so it is not part of the request to SetValue.
+
+            // Unchanged package source mapping.
+            Dictionary<string, object> packageSourceDictionary3 = new Dictionary<string, object>();
+            packageSourceDictionary3[PackageSourceMappingPage.MonikerPackageId] = unitTestingSourceMapping3.Patterns.Single().Pattern;
+            packageSourceDictionary3[PackageSourceMappingPage.MonikerSourceNames] = new List<string>() { unitTestingSourceMapping3.Key };
+
+            // Add a new package source mapping for another existing source.
+            Dictionary<string, object> packageSourceDictionary4 = new Dictionary<string, object>();
+            packageSourceDictionary4[PackageSourceMappingPage.MonikerPackageId] = packageIdPattern;
+            packageSourceDictionary4[PackageSourceMappingPage.MonikerSourceNames] = new List<string>() { addNewPackageSourceName4 };
+
+            IList<IDictionary<string, object>> sourceMappingDictionaryList =
+                new List<IDictionary<string, object>>(capacity: 3)
+                {
+                    packageSourceDictionary1, // Pre-existing and unchanged.
+                    // No source mapping is sent by Unified Settings for nonExistantSourceName2.
+                    packageSourceDictionary3, // Pre-existing and unchanged.
+                    packageSourceDictionary4 // A newly added source mapping.
+                };
+
+            // Act
+            ExternalSettingOperationResult result = await instance.SetValueAsync(
+                PackageSourceMappingPage.MonikerPackageSourceMapping,
+                sourceMappingDictionaryList,
+                CancellationToken.None);
+
+            // Assert
+            result.Should().NotBeNull();
+            result.Should().BeOfType<ExternalSettingOperationResult.Success>();
+
+            IReadOnlyList<PackageSourceMappingSourceItem> packageSourceMappingItems = instance._packageSourceMappingProvider.GetPackageSourceMappingItems();
+
+            // If the machine-wide config contains package source mappings, this count could be greater than 4.
```

Comment:
Is this the config on the machine running the tests?

Reply:
Yes - I'm searching for a better test utility that will not use the machine's config for testing. I reached out to Andy yesterday for opinions. NuGet.CommandLine.Test has a Util to create a variety of configs... not sure why it's embedded in CommandLine. May work well as a general Util.
The test doesn't really care, it just asserts that there's at least 4. The comment is saying the test doesn't fail if there's more from a machine-wide config.
https://github.com/NuGet/NuGet.Client/blob/97383e8d69f0add4bcdf937458047960c9b9f8f0/test/NuGet.Clients.Tests/NuGet.CommandLine.Test/Util.cs#L462

Reply:
There's a lot of utilities available.

https://github.com/NuGet/NuGet.Client/blob/dev/test/TestUtilities/Test.Utility/SimpleTestSetup/SimpleTestPathContext.cs
https://github.com/NuGet/NuGet.Client/blob/dev/test/TestUtilities/Test.Utility/SimpleTestSetup/SimpleTestSettingsContext.cs

> The comment is saying the test doesn't fail if there's more from a machine-wide config.

Yeah, the comment made sense, but the flaw is that we're being impacted by the machine wide config.